neptune_optuna/impl/__init__.py

```diff
@@ -199,15 +211,52 @@ def _should_log_plots(self, study: optuna.study.Study, trial: optuna.trial.Froze
     def _should_log_study(self, trial: optuna.trial.FrozenTrial):
         if self._study_update_freq == 'never':
             return False
-        else:
-            if trial._trial_id % self._study_update_freq == 0:
-                return True
+        if trial._trial_id % self._study_update_freq == 0:
+            return True
         return False
 
+    def log_best_trials(self, study: optuna.Study):
+        _log_best_trials(self.run, study, namespaces=self._namespaces)
+
+def _log_best_trials(run, study: optuna.Study, namespaces: List[str] = None):
+    if study._is_multi_objective():
+        if namespaces is None:
+            namespaces = get_namespaces(study)
+        _log_trials(run, study, trials=study.best_trials, namespaces=namespaces, best=True)
+    else:
+        _log_single_trial(run, study, trial=study.best_trial, namespaces=namespaces, best=True)
+
+
+def get_namespaces(
+    study: optuna.Study,
+    target_names: List[str] = None
+    ) -> Optional[List[str]]:
+
+    if study._is_multi_objective():
+        if target_names is None:
+            return list(map(lambda direction_index: f'objective_{direction_index}', range(len(study.directions))))
+
+        assert len(target_names) == len(study.directions), \
+            """
+            The target_names list must be th same length as study.directions.
+            target_names length: {} != study.directions length: {}
+            """.format(len(target_names), len(study.directions))
+        return target_names
+    else:
+        if target_names is None:
+            return 'Objective Value'
```

Comment:
I think that this should follow lowercase, `-` delimited format (it doesn't match for the multi-objective case)